src/GUI/UI/UIGameOver.cpp

```diff
@@ -50,8 +46,15 @@ void GUIWindow_GameOver::Release() {
     engine->_messageQueue->addMessageCurrentFrame(_releaseEvent, 0, 0);
 
     current_screen_type = prev_screen_type;
-    bGameoverLoop = false;
+    GameOverNoSound = false;
     pEventTimer->setPaused(false);
 
     GUIWindow::Release();
 }
+
+bool GUIWindow_GameOver::isFinished() {
+    if (_showPopUp) return true;
+
+    _showPopUp = true;
```

Comment:
This looks off to me... My expectation for `isFinished` is that it should be `const` and non-mutating.

Reply:
Yeah thats a very poor naming choice